frontend/src/pages/CreatePoll.js

```diff
@@ -0,0 +1,104 @@
+import Page from "../components/Page";
+import Header from "../components/Header";
+import TextField from "../components/TextField";
+import {useState} from "react";
+import Button from "../components/Button";
+import {createPoll} from "../service/api-service";
+import Loading from "../components/Loading";
+import Error from "../components/Error";
+import AnswerList from "../components/AnswerList";
+import Navbar from "../components/Navbar";
+import styled from "styled-components/macro";
+import {useAuth} from "../auth/AuthProvider";
+import {Redirect} from "react-router-dom";
+
+export default function CreatePoll() {
+
+    const {user, token} = useAuth()
+    const [pollTitle, setPollTitle] = useState('')
+    const [answerToAdd, setAnswerToAdd] = useState('')
+    const [possibleAnswers, setPossibleAnswers] = useState([])
+    const [loading, setLoading] = useState(false)
+    const [error, setError] = useState()
+
+    const handleSubmit = event => {
+        const poll = {
+            title: pollTitle,
+            user: user,
+            possibleAnswers: possibleAnswers
+        }
+        event.preventDefault()
+        setError()
+        setLoading(true)
+        createPoll(poll, token).catch(error => {
+            setError(error)
+        })
+        setLoading(false)
```

Comment:
You want to change this after the promise form `createPoll` has settled. Do it in a `then`, `catch` or `finally`.